--- a/net/minecraft/entity/passive/AbstractHorse.java
+++ b/net/minecraft/entity/passive/AbstractHorse.java
@@ -1167,7 +1167,7 @@
 
     public void updatePassenger(Entity passenger)
     {
-        super.updatePassenger(passenger);
+        
 
         if (passenger instanceof EntityLiving)
         {
@@ -1187,7 +1187,7 @@
             {
                 ((EntityLivingBase)passenger).renderYawOffset = this.renderYawOffset;
             }
-        }
+        } super.updatePassenger(passenger);
     }
 
     /**
