--- a/net/minecraft/client/particle/ParticleFirework.java
+++ b/net/minecraft/client/particle/ParticleFirework.java
@@ -19,9 +19,11 @@
     {
         public Particle createParticle(int particleID, World worldIn, double xCoordIn, double yCoordIn, double zCoordIn, double xSpeedIn, double ySpeedIn, double zSpeedIn, int... p_178902_15_)
         {
-            ParticleFirework.Spark particlefirework$spark = new ParticleFirework.Spark(worldIn, xCoordIn, yCoordIn, zCoordIn, xSpeedIn, ySpeedIn, zSpeedIn, Minecraft.getMinecraft().effectRenderer);
-            particlefirework$spark.setAlphaF(0.99F);
-            return particlefirework$spark;
+        	ParticleFirework.Spark particlefirework$spark = new ParticleFirework.Spark(worldIn, xCoordIn, yCoordIn, zCoordIn, xSpeedIn, ySpeedIn, zSpeedIn, Minecraft.getMinecraft().effectRenderer);
+        	particlefirework$spark.setAlphaF(0.99F);
+        	if(p_178902_15_.length >=3)
+        		particlefirework$spark.setRBGColorF(p_178902_15_[0]/255f, p_178902_15_[1]/255f,p_178902_15_[2]/255f);
+        	return particlefirework$spark;
         }
     }
 
